crates/kubelet/src/config.rs

```diff
@@ -178,6 +193,13 @@ pub struct Opts {
     )]
     node_ip: Option<IpAddr>,
 
+    #[structopt(
+        long = "node-labels",
+        env = "NODE_LABELS",
+        help = "Labels to add when registering the node in the cluster. Labels must be key-value pairs separated by ','"
+    )]
+    node_labels: Option<String>,
```

Comment:
It might be valuable to use a repeated flag instead. That way users can specify node labels like `--node-labels foo=bar --node-labels car=star` rather than one looooong string. It also removes all of the comma splitting logic.

StructOpt also has the option to provide a delimiter, so your example would work the same with

```
#[structopt(long = "node-labels", raw(use_delimiter = "true"))]
node_labels: Vec<String>
```

source: https://github.com/TeXitoi/structopt/issues/94

Reply:
There's incentive to keep it the same way that `kubelet` works, primarily that tools like `eksctl` generate an environment file for the service with `NODE_LABELS` set to a comma-delimited value.

It would be nice to simply pick these values up directly from the tooling without having to special-case it for krustlet (i.e. read the value that got generated by `eksctl` and transform it in bootstrapping to something `krustlet` supports).

Reply:
Oh, I missed the point that StructOpt would still support the delimiter. :+1: then!

Reply:
Good call. I changed it to `use_delimiter = true` in https://github.com/deislabs/krustlet/pull/197/commits/f71f6e1f039710716127d1c38216c8b352b4e0c7. FWIW, `raw` was removed and it becomes https://github.com/TeXitoi/structopt/issues/367#issuecomment-606619645 now.